common/src/main/java/com/wynntils/models/gear/SetModel.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.gear;
+
+import com.google.gson.reflect.TypeToken;
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.components.Model;
+import com.wynntils.core.components.Models;
+import com.wynntils.core.net.Download;
+import com.wynntils.core.net.UrlId;
+import com.wynntils.models.stats.type.StatType;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SetModel extends Model {
+    private final Map<String, SetInfo> setData = new HashMap<>();
+
+    public SetModel() {
+        super(List.of());
+        loadSetData();
+    }
+
+    public SetInfo getSetData(String setId) {
+        return setData.get(setId);
+    }
+
+    private void loadSetData() {
+        Download dl = Managers.Net.download(UrlId.DATA_STATIC_ITEM_SETS);
+        dl.handleReader(reader -> {
+            Map<String, Map<String, List<?>>> tempData =
+                    WynntilsMod.GSON.fromJson(reader, new TypeToken<Map<String, Map<String, List<?>>>>() {}.getType());
+            tempData.forEach((key, value) -> {
```

Comment:
Why not create a class that represents the JSON value?

Reply:
We often use private "decoder" classes so that we don't need to use such big type definitions, and GSON can still deserialize the data.